python/paddle/fluid/tests/unittests/CMakeLists.txt

```diff
@@ -23,7 +23,9 @@ if(NOT WITH_DISTRIBUTE)
     LIST(REMOVE_ITEM TEST_OPS test_dist_text_classification)
 endif(NOT WITH_DISTRIBUTE)
 
-if (${CUDNN_MAJOR_VERSION} VERSION_LESS 7)
+if (NOT ${WITH_GPU})
+    LIST(REMOVE_ITEM TEST_OPS test_conv2d_fusion_op)
+elseif(${CUDNN_MAJOR_VERSION} VERSION_LESS 7)
```

Comment:
Fix a CPU compile error by the way.  The `CUDA_MAJOR_VERSION` is not defined when `WITH_GPU` is OFF. The `if` statement will fire a fatal error.

We can not use `if (not WITH_GPU OR CUDA_MAJOR_VERSION  VERSION_LESS 7)` since the `cmake` does not seem to have short-cut feature in `if` statement.

Reply:
I encounter the same cmake error @qingqing01 
```
CMake Error at python/paddle/fluid/tests/unittests/CMakeLists.txt:26 (if):
  if given arguments:

    "VERSION_LESS" "7"

  Unknown arguments specified


-- Configuring incomplete, errors occurred!
```

Reply:
@reyoung In current CI, we don't compile `WITH_TESTING=ON` in only-CPU environment, should we add this test case?

Reply:
@luotao1 I think Mac ci can cover this condition

Reply:
How about simpily to :
```
if (NOT (${WITH_GPU} AND ${CUDNN_MAJOR_VERSION} VERSION_GREATER 7))
```